src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,16 +38,29 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private formlyJsonschema: FormlyJsonschema
   ) {}
 
   ngOnInit(): void {
+    this.action = 'Create';
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
```

Comment:
* This should be in a separate service. Components should be lean and removed from any data processing logic.
* Nested subscriptions is an anti-pattern. Use mergeMap/switchMap/etc., instead.
* Use subscribe only to assign things, and ideally don't use subscribe at all. Just assign the output of the `.pipe()` expression to a `variable$`, and use the `*ngIf="variable$ | async as variable"` in the HTML template.

https://blog.brecht.io/rxjs-best-practices-in-angular/

Reply:
I fixed this. Used mergeMap and async pipe instead.